Fix PATH in setup_KWIVER.bat

mleotta opened the pull request:
This fix was needed for me to package MAP-Tk.  It seems like the right thing to do, but I'm not sure if there was some other reason we were intentionally replacing rather than appending to PATH.

mleotta commented:
That all makes sense.  I would not be worried about calling the script many times, I'm not sure why anyone would need to do that.  The only potential problem with putting path at the end is that we have other lines that get appended to this file that also set the PATH (I'd have to track down where they come from).  Those lines have the form `set PATH=%PATH%;other_stuff`.  So that "other_stuff" would still come after the system paths.  Or if we changed those commands to post-fix PATH as well, those additional paths would now come before the previous paths we added, not after as they do now.

I'm not sure if any of this path ordering will cause a problem.  I'm just pointing it out.

aaron-bray commented:
Yep

In the kwiver root CMakeLists.txt  would change those too to

Line 222:   file( APPEND "${KWIVER_SETUP_BATCH_FILE}" "set PATH=%PATH%;${fletch_ROOT}/bin\n" )
	Line 223:   file( APPEND "${KWIVER_SETUP_BATCH_FILE}" "set PATH=%PATH%;${fletch_ROOT}/x64/${_vcVersion}/bin\n" )
	Line 276:   file( APPEND "${KWIVER_SETUP_BATCH_FILE}" "set PATH=%PATH%;${Matlab_LIBRARY_DIR}\n" )

That is it for PATH set up in kwiver, should be good with those changes

mleotta commented:
So the side effect of that proposed change is that the order of those KWIVER paths is now reversed from what it was before.  For example, the MATLAB_library_DIR would be first on the PATH instead of last as it is now, but all of our added paths would be before all system paths.  This may not be a problem at all, I just want to confirm that no one sees any obvious issues with this.

And yes, I could reorder those append lines to partially correct for this, but still, all the appended lines will add things before the line that's hard coded in the file, which is different than it is now.

aaron-bray commented:
All the paths we will prepend to the path are our various kwiver and fletch build paths. 

- kiwvier modules/bin/sprokit/lib
- fletch bin
- fletch ocv bin
- matlab bin?(never build this before...)

I don't feel the order of those will matter as long as they are all before the current PATH
I don't think there is any dll of the same name that are different in these directories that could cause some kind of issue (if we do, that should be fixed...)

mleotta commented:
I hadn't addressed the feedback from @aaron-bray yet.  That said, this change fixes a problem as is, and doesn't seem to make things worse, so no need to back out the merge.  I will just follow up with the PATH ordering switch.

dstoup commented:
Sorry about that. It looked like Aaron had implicitly approved the PR.

On Mon, Jul 30, 2018 at 8:40 AM, Matt Leotta <notifications@github.com>
wrote:

> I hadn't addressed the feedback from @aaron-bray
> <https://github.com/aaron-bray> yet. That said, this change fixes a
> problem as is, and doesn't seem to make things worse, so no need to back
> out the merge. I will just follow up with the PATH ordering switch.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/kwiver/pull/580#issuecomment-408849368>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFRAWvb6EwKDCakZvyhCfPRkrj0DXCQcks5uLv7BgaJpZM4VjuzE>
> .
>



-- 
David Stoup
Principal Engineer

Kitware, Inc.
28 Corporate Drive
Clifton Park, NY. 12065
518-881-4949 (W)
518-312-3946 (M)
518-371-4573 (F)
